spec/workflow/intrinsic_function_spec.rb

```diff
@@ -103,9 +103,10 @@
         expect(result).to eq(["bar"])
       end
 
+      # TODO: test this
       it "handles invalid path references" do
-        result = described_class.value("States.Array($.xxx)", {"context" => {"baz" => "qux"}}, {"input" => {"foo" => "bar"}})
-        expect(result).to eq([nil])
+        ctx = {"context" => {"baz" => "qux"}}, {"input" => {"foo" => "bar"}}
+        expect { described_class.value("States.Array($.xxx)", ctx) }.to raise_error(Floe::PathError, "references an invalid value")
```

Comment:
ok, verified this:

<details>
<summary>asl file with tests</summary>

```asl
{
  "Comment": "Testing Pass intrinsics - sample input: {test: 1}",
  "StartAt": "Setup",
  "States": {
    "Setup": {
      "Comment": "(Implied test) Result => output:Result",
      "Type": "Pass",
      "Next": "ChooseTest",
      "Parameters": {"test.$": "$.test", "num1": 1, "str1": "1", "num2": 1, "null": null, "str2": "1"}
    },
    "ChooseTest": {
      "Type":  "Choice",
      "Choices": [
        {"Variable": "$.test", "NumericEquals": 1, "Next": "Test1"},
        {"Variable": "$.test", "NumericEquals": 2, "Next": "Test2"}
      ],
      "Default": "Default"
    },
    "Test1": {
      "Comment": "Variable missing -> The function 'States.Array($.missing)' had the following error: The JsonPath argument for the field '$.missing' could not be found in the input ...",
      "Type": "Pass",
      "Parameters": {
        "param1.$": "States.Array($.missing)"
      },
      "Next": "Match"
    },
    "Test2": {
      "Comment": "output: {param1: [null]}",
      "Type": "Pass",
      "Parameters": {
        "param1.$": "States.Array($.null)"
      },
      "Next": "Verify2"
    },
    "Verify2": {
      "Type": "Choice",
      "Choices": [
        {
          "Or": [
            {"Variable": "$.param1[0]", "IsPresent": true},
            {"Variable": "$.param1[0]", "IsNull": true},
            {"Variable": "$.param1[1]", "IsPresent": false}
          ], "Next": "Match"
        }
      ],
      "Default": "Default"
    },
    "Match": {
      "Type": "Pass",
      "End": true
    },
    "Default": {
      "Type": "Pass",
      "End": true,
      "Result": "Nothing matched"
    }
  }
}
```

</details>

Reply:
yeah that fails in the simulator - can you verify what a path does without the intrinsic function but with a missing key?  If that also fails then I think we should fail this in the Path type.